src/HoneybadgerServiceProvider.php

```diff
@@ -125,6 +125,36 @@ private function registerEventHooks()
                 }
             });
         });
+
+        /** @param string|array|null $environments */
+        Event::macro('thenPingHoneybadgerFromConfig', function (string $configKey, $environments = null) {
+            $id = config($configKey);
+
+            if (! $id) {
+                return $this;
+            }
+
+            return $this->then(function () use ($id, $environments) {
+                if ($environments === null || app()->environment($environments)) {
+                    app(Reporter::class)->checkin($id);
+                }
+            });
```

Comment:
This block of code is the same as L146-156. Can you extract into a function that can be called from both macros?

Reply:
Shoot, I had actually done that and just forgot to commit it - makes total sense.